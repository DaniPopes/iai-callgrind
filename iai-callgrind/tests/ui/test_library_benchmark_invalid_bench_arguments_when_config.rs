use iai_callgrind::library_benchmark;

#[library_benchmark]
#[bench::id(config = "some")]
pub fn bench0() {}

#[library_benchmark]
#[bench::id(wrong = LibraryBenchmarkConfig::default())]
pub fn bench1() {}

#[library_benchmark]
#[bench::id(config = )]
pub fn bench2() {}

fn main() {}
