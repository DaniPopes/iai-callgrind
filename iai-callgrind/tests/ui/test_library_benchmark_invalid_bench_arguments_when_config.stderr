error: Invalid argument: wrong

         = help: Valid arguments are: args, config

 --> tests/ui/test_library_benchmark_invalid_bench_arguments_when_config.rs:8:13
  |
8 | #[bench::id(wrong = LibraryBenchmarkConfig::default())]
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unexpected end of input, expected an expression
  --> tests/ui/test_library_benchmark_invalid_bench_arguments_when_config.rs:12:22
   |
12 | #[bench::id(config = )]
   |                      ^

error[E0277]: the trait bound `RunnerLibraryBenchmarkConfig: From<&str>` is not satisfied
 --> tests/ui/test_library_benchmark_invalid_bench_arguments_when_config.rs:3:1
  |
3 | #[library_benchmark]
  | ^^^^^^^^^^^^^^^^^^^^ the trait `From<&str>` is not implemented for `RunnerLibraryBenchmarkConfig`
  |
  = help: the following implementations were found:
            <RunnerLibraryBenchmarkConfig as From<&LibraryBenchmarkConfig>>
            <RunnerLibraryBenchmarkConfig as From<&mut LibraryBenchmarkConfig>>
            <RunnerLibraryBenchmarkConfig as From<LibraryBenchmarkConfig>>
  = note: required because of the requirements on the impl of `Into<RunnerLibraryBenchmarkConfig>` for `&str`
  = note: this error originates in the attribute macro `library_benchmark` (in Nightly builds, run with -Z macro-backtrace for more info)
